torch/csrc/generic/StorageSharing.cpp

```diff
@@ -233,9 +248,38 @@ static PyObject * THPStorage_(shareCuda)(THPStorage *self)
 
     _handle = PyBytes_FromStringAndSize((char *)&handle, CUDA_IPC_HANDLE_SIZE);
     _offset_bytes = PyLong_FromSsize_t((Py_ssize_t)offset_bytes);
+
+    if (!torch::CudaIPCHaveRefCounter())
+    {
+      int flags = TH_ALLOCATOR_MAPPED_SHAREDMEM | TH_ALLOCATOR_MAPPED_EXCLUSIVE;
+      std::string ref_counter_handle = THPStorage_(__newHandle)();
+      at::DataPtr sptr = THRefcountedMapAllocator::makeDataPtr(
+          ref_counter_handle.c_str(), flags, sizeof(int64_t) * torch::CUDA_IPC_REF_COUNTER_FILE_SIZE, nullptr);
+      torch::CudaIPCCreateRefCounter(ref_counter_handle, torch::CUDA_IPC_REF_COUNTER_FILE_SIZE, std::move(sptr));
+    }
+
+    // Put Storage Data behind new ref counting context
+    at::DataPtr sent_data_ptr = torch::GetNewRefCountedSentData(storage->data(), storage->device());
+    auto old_data_ptr = storage->set_data_ptr(std::move(sent_data_ptr));
+    auto sent_data  =  static_cast<torch::CudaIPCSentData*>(storage->data_ptr().get_context());
+    sent_data->set_original_ptr(std::move(old_data_ptr));
+    _ref_counter = PyBytes_FromString((sent_data->handle()).c_str());
+    _ref_counter_offset = PyLong_FromLong(sent_data->offset());
+
+
+    cudaIpcEventHandle_t ipc_event_handle;
+
+#ifndef __HIP_PLATFORM_HCC__
+    THCudaCheck(cudaIpcGetEventHandle(&ipc_event_handle, sent_data->event_));
+#else
+    // Pass any uninitialized data, we are not going to use it.
```

Comment:
I did a double take on this comment; it took me a moment to realize you meant "pass any uninitialized data in `ipc_event_handle`"

Reply:
A little note saying, `hipIpcGetEventHandle` doesn't exist, which is why we ifdef'ed this would help. Also, IIUC, for the HIP case we're just not going to synchronize at all, is that right? Also worth pointing out.

Reply:
clarified